test/vfd_swmr_attrdset_writer.c

```diff
@@ -0,0 +1,2015 @@
+#define NP_INITIALIZER (np_state_t) {		                        \
+	  .fifo_writer_to_reader = "./fifo_attrdset_writer_to_reader"   \
+	, .fifo_reader_to_writer = "./fifo_attrdset_reader_to_writer"   \
+	, .fd_writer_to_reader = -1                                     \
+	, .fd_reader_to_writer = -1							            \
+    , .notify = 0                                                   \
+    , .verify = 0                                                   }
+
+static bool state_init(state_t *, int, char **);
+
+static bool np_init(np_state_t *np, bool writer);
+static bool np_close(np_state_t *np, bool writer);
+static bool np_writer(bool result, unsigned step, const state_t *s,  np_state_t *np, H5F_vfd_swmr_config_t *config);
+static bool np_reader(bool result, unsigned step, const state_t *s, np_state_t *np);
+static bool np_confirm_verify_notify(int fd, unsigned step, const state_t *s, np_state_t *np);
+static bool np_reader_no_verification(const state_t *s, np_state_t *np,  H5F_vfd_swmr_config_t *config);
+
+static bool create_dsets(const state_t *s, dsets_state_t *ds);
+static bool open_dsets(const state_t *s, dsets_state_t *ds);
+static bool open_dset_real(hid_t fid, hid_t *did, const char *name, unsigned *max_compact, unsigned *min_dense);
+static bool close_dsets(const dsets_state_t *ds);
+
+static bool attr_dsets_action(unsigned action, const state_t *s, const dsets_state_t *ds, unsigned step);
+static bool attr_action(unsigned action, const state_t *s, hid_t did, unsigned step);
+static bool add_attr(const state_t *s, hid_t did, unsigned int which);
+static bool modify_attr(const state_t *s, hid_t did, unsigned int which);
+static bool delete_attr(hid_t did, unsigned int which);
+
+static bool verify_attr_dsets_action(unsigned action, const state_t *s, const dsets_state_t *ds, unsigned which);
+static bool verify_attr_action(unsigned action, hid_t did, unsigned which);
+static bool verify_add_or_modify_attr(unsigned action, hid_t did, char *attr_name, unsigned int which);
+static bool verify_delete_attr(hid_t did, char *attr_name);
+static bool verify_storage_cont(unsigned action, hid_t did, unsigned int which, unsigned max_compact, unsigned min_dense, unsigned asteps);
+static bool verify_storage_cont_real(hid_t did, unsigned int which, unsigned cut_point);
+
+static const hid_t badhid = H5I_INVALID_HID;
+
+/* Names for datasets */
+#define DSET_COMPACT_NAME   "compact_dset"
+#define DSET_CONTIG_NAME    "contig_dset"
+#define DSET_SINGLE_NAME    "chunked_single"
+#define DSET_IMPLICIT_NAME  "chunked_implicit"
+#define DSET_FA_NAME        "chunked_fa"
+#define DSET_EA_NAME        "chunked_ea"
+#define DSET_BT2_NAME       "chunked_bt2"
+
+/* Action for attribute handling */
+#define ADD_ATTR        1
+#define MODIFY_ATTR     2
+#define DELETE_ATTR     3
+
+/* This will be removed when this routine is moved to vfd_swmr_common.c */
+/* Sleep for `tenths` tenths of a second */
+static void
+decisleep(uint32_t tenths)
```

Comment:
I put the definition of this function in `vfd_swmr_common.c` in my pull request for the zoo test.  I already merged my PR.

Reply:
Corrected.